packages/graphql-typescript-definitions/src/index.ts

```diff
@@ -254,7 +259,33 @@ export class Builder extends EventEmitter {
     });
   }
 
+  private isDocumentTypesJobCancelled(jobId: number) {
+    return this.activeDocumentTypesJobId !== jobId;
+  }
+
   private async generateDocumentTypes() {
+    // Document generation can be called often via:
+    // – Projects with small + large schemas calling `handleSchemaUpdate`
+    //   during schema refreshes
+    // - Projects with many .graphql triggering multiple `handleDocumentUpdate`
+    //   calls during rebases
+    //
+    // The above scenarios do not provide a predictable debounce window that
+    // would prevent multiple document generation jobs running, so generation
+    // is allowed to immediately proceed, but each "job" checks itself often to
+    // see if it should stop.
+    const jobId = ++this.activeDocumentTypesJobId;
```

Comment:
This is scrappy 😓  It'd be neat to turn `_generateDocumentTypes` into a generator that returns jobs to execute. Happy to noodle on something like that, but I'd need a push in that direction 😸

Reply:
Is there anyway this could collide as this is just an `int`?